--- include/wx/msw/missing.h	Thu Jun 02 17:19:23 2005
+++ include/wx/msw/missing.h	Sun Sep 18 17:46:02 2005
@@ -16,4 +16,21 @@
  */
 
+#ifndef HANGUL_CHARSET
+    #define HANGUL_CHARSET  129
+#endif
+
+#ifndef CCM_SETUNICODEFORMAT
+    #define CCM_SETUNICODEFORMAT 8197
+#endif
+
+#ifndef TVS_FULLROWSELECT
+    #define TVS_FULLROWSELECT       0x1000
+#endif
+
+#ifndef TVM_SETBKCOLOR
+    #define TVM_SETBKCOLOR          (TV_FIRST + 29)
+    #define TVM_SETTEXTCOLOR        (TV_FIRST + 30)
+#endif
+
 // Needed by cursor.cpp
 #ifndef IDC_HAND
