Index: build/msw/config.vc
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/build/msw/config.vc,v
retrieving revision 1.19
diff -u -2 -r1.19 config.vc
--- build/msw/config.vc	2005/07/28 12:42:10	1.19
+++ build/msw/config.vc	2005/09/18 13:13:46
@@ -27,5 +27,5 @@
 
 # Standard linker flags 
-LDFLAGS = 
+LDFLAGS = /incremental:no
 
 # What type of library to build? [0,1]
@@ -57,3 +57,3 @@
-DEBUG_RUNTIME_LIBS = default
+DEBUG_RUNTIME_LIBS = 0
 
 # Multiple libraries or single huge monolithic one? [0,1]
