--- src/common/intl.cpp	Tue Sep 20 15:35:21 2005
+++ src/common/intl.cpp	Sun Oct 30 20:51:32 2005
@@ -1260,9 +1260,14 @@
             inputConv =
             csConv = new wxCSConv(m_charset);
     }
-    else // no conversion needed
+    else // no need to convert the encoding
     {
+        // we still need the conversion for Unicode build
+#if wxUSE_UNICODE
+        inputConv = wxConvCurrent;
+#else // !wxUSE_UNICODE
         inputConv = NULL;
+#endif
     }
 
     // conversion to apply to msgid strings before looking them up: we only
